ddtrace/tracer.py

```diff
@@ -400,6 +401,35 @@ def configure(
                     msg = "- DATADOG TRACER DIAGNOSTIC - %s" % agent_error
                     self._log_compat(logging.WARNING, msg)
 
+    def _child_after_fork(self):
+        self._pid = getpid()
+
+        # Assume that the services of the child are not necessarily a subset of those
+        # of the parent.
+        self._services = set()
+
+        # Re-create the background writer thread
+        self.writer = self.writer.recreate()
+        self._initialize_span_processors()
+
+        # The spans remaining in the context can not and will not be
+        # finished in this new process. So to avoid memory leaks the
+        # strong span reference (which will never be finished) is replaced
+        # with a context representing the span.
+        child_of = self.context_provider.active()
+        if isinstance(child_of, Span):
+            new_ctx = Context(
+                sampling_priority=child_of.context.sampling_priority,
+                span_id=child_of.span_id,
+                trace_id=child_of.trace_id,
+            )
+
+            # If the child_of span was active then activate the new context
+            # containing it so that the strong span referenced is removed
+            # from the execution.
+            if self.context_provider.active() is child_of:
+                self.context_provider.activate(new_ctx)
```

Comment:
do we need a lock around this? the double call to `active()` to check if the span we are trying to modify is still active smells like a race condition ?

should the context provider register a forksafe hook and handle this there instead?

Reply:
> do we need a lock around this?

FWIW at this stage, there can't be any other thread running so we should be good.
If there was any other thread, it'd be one the tracer/profiler started at this stage. No "userland" thread.

Reply:
ok, that is what I figured would be the case, since this runs right after forking.

given this, do we need the additional check on or can we just `activate(new_ctx)`?

maybe a good question for @Kyle-Verhoog

Reply:
lgtm :+1:! (this is the same code as before just pulled out into this hook)